demos/linear-wave-equation/linear_wave_equation.py.rst

```diff
@@ -0,0 +1,148 @@
+Double slit experiment
+======================
+
+Here we solve a linear wave equation using an explicit timestepping
+scheme. This example demonstrates the use of an externally generated
+mesh, pointwise operations on Functions, and a time varying boundary
+condition. The strong form of the equation we set out to solve is:
+
+.. math::
+
+   \frac{\partial^2\phi}{\partial t^2} - \nabla^2 \phi = 0
+
+   \nabla \phi \cdot n = 0 \ \textrm{on}\ \Gamma_N
+   
+   \phi = \frac{1}{10\pi}\cos(10\pi t)  \ \textrm{on}\ \Gamma_D
+
+To facilitate our choice of time integrator, we make the substitution:
+
+.. math::
+   
+   \frac{\partial\phi}{\partial t} = - p
+
+   \frac{\partial p}{\partial t} + \nabla^2 \phi = 0
+
+   \nabla \phi \cdot n = 0 \ \textrm{on}\ \Gamma_N
+   
+   p = \sin(10\pi t)  \ \textrm{on}\ \Gamma_D
+
+We then form the weak form of the equation for :math:`p`. Find
+:math:`p \in V` such that:
+
+.. math::
+
+   \int_\Omega \frac{\partial p}{\partial t} v\,\mathrm d x = \int_\Omega \nabla\phi\cdot\nabla v\,\mathrm d x
+   \quad \forall v \in V
+
+For a suitable function space V. Note that the absence of spatial
+derivatives in the equation for :math:`\phi` makes the weak form of
+this equation equivalent to the strong form so we will solve it pointwise.
+
+In time we use a simple symplectic method in which we offset :math:`p`
+and :math:`\phi` by a half timestep.
+
+This time we created the mesh with `Gmsh <http://geuz.org/gmsh/>`_:
+
+.. code-block:: bash
+
+   gmsh -2 wave_tank.msh
+
+We can then start our Python script and load this mesh::
+
+  from firedrake import *
+  mesh = Mesh("wave_tank.msh")
+
+We choose a degree 1 continuous function space, and set up the
+function space and functions::
+
+  V = FunctionSpace(mesh, 'Lagrange', 1)
+  p = Function(V, name="p")
+  phi = Function(V, name="phi")
```

Comment:
It might be nice to say that the `name` parameter is used to name the field in the output file (assuming that *is* what it does?)